File: atcoder/abc141B_tap_dance.ml
(* Vicfred
 * https://atcoder.jp/contests/abc141/tasks/abc141_b
 * implementation
 * *)
Scanf.scanf "%s\n" (fun s ->
    if
      Array.fold_left ( && ) true @@
      Array.init (String.length s) @@ fun i ->
        if i mod 2 = 0 && s.[i] = 'L' then false
        else if i mod 2 = 1 && s.[i] = 'R' then false
        else true
    then "Yes"
    else "No"
) |> Printf.printf "%s\n"

File: atcoder/abc141B_tap_dance.lua
-- unihernandez22
-- https://atcoder.jp/contests/abc141/tasks/abc141_b
-- implementation

s = io.read()
ans = "Yes"

for i = 1, s:len(), 1 do
  if i % 2 == 1 and
       s:sub(i,i) ~= 'R' and s:sub(i,i) ~= 'U' and
       s:sub(i,i) ~= 'D' or 
     i % 2 == 0 and
       s:sub(i,i) ~= 'L' and s:sub(i,i) ~= 'U' and
       s:sub(i,i) ~= 'D' then
    ans = "No"
    break
  end
end

print(ans)
